File: C_Sharp/Framework.Tests/LazyLoad/ProductDomain.cs
using Framework.Data_Manipulation;
using Framework.LazyLoad;

namespace Framework.Tests.LazyLoad
{
    public class ProductDomain : LazyLoadDomainObject<ProductDomain.Criteria>
    {
        public class Criteria
        {
            public int Id { get; set; }

            public static Criteria SearchById(int id)
            {
                return new Criteria {Id = id};
            }
        }

        public int Id { get; set; }
        public string Description { get; set; }

        public ProductDomain(IBaseMapper mapper, Criteria criteria)
            : base(criteria, mapper)
        {
            if (criteria != null)
                Id = criteria.Id;
        }
    }
}

File: Android/app/src/test/java/com/peaa/LazyLoad/ProductDomain.java
package com.peaa.LazyLoad;

import com.peaa.DataManipulation.BaseMapperInterfaces.IBaseMapper;

/**
 * Created by aiko on 4/8/17.
 */

public class ProductDomain extends LazyLoadDomainObject<ProductDomain.Criteria>{
    public static class Criteria
    {
        public int Id;

        public Criteria(int id)
        {
            Id = id;
        }
    }

    public int Id;
    public String Description;

    public ProductDomain(IBaseMapper mapper, Criteria criteria)
    {
        super(criteria, mapper);

        if (criteria != null)
            Id = criteria.Id;
    }
}
